airframe/src/main/scala/wvlet/airframe/lifecycle/LifeCycleManager.scala

```diff
@@ -310,6 +332,13 @@ object FILOLifeCycleHookExecutor extends LifeCycleEventHandler with LogSupport {
     }
   }
 
+  override def afterStart(lifeCycleManager: LifeCycleManager): Unit = {
```

Comment:
I guess `afterStart` comes from after `onStart`. However, it's hard to choose an appropriate method by considering which methods are already used (e.g. I have to use `afterStart` because `onStart` is already used). Rather, it might be better to give a name which expresses the purpose of the method. In this case, a name which expresses that this method should be used for customizing a component. I thought `onInit` is better for customization, but it already exists... 😓 

Or, just explaining the purpose of this life cycle method on Scaladoc or README might be enough.

Reply:
It's a valid concern, so I'd like to describe the mode details about the appropriate use case of  `afterStart` (and `beforeShutdown`).

The lifecycle phases are:
onInit (instance initialization) -> onInject (when DI is injecting instance, can be called multiple times) -> onStart (when session starts) -> afterStart (after onStart, use this only for customizing test design) -> beforeShutdown -> onShutdown 

Currently, `.onInit` is used for _instance_ initialization _before_ the injection regardless of the session state (not started or started).